Source/SimCacheModule/Source/Events/UIRangeAnnulusChangedEvent.h

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) 2024 Steven Frost and Orion Lyau
+
+#pragma once
+
+#include "Core/RangeAnnulus.h"
+
+#include "Events/RangeAnnulusChangedEvent.h"
```

Comment:
Pretty sure?  We have this constructor to create a `UIRangeAnnulusChangedEvent` from a `RangeAnnulusChangedEvent` (and `RangeAnnulusChangedEvent` is defined in Events/RangeAnnulusChangedEvent.h?):

```cpp
UIRangeAnnulusChangedEvent( const RangeAnnulusChangedEvent& Event )
	: Annulus( Event.Annulus )
{
}
```

Reply:
Ah sorry I missed that

Reply:
No worries!  Though, I guess if we _do_ want to remove the dependency -- do you think it would be better to change that constructor to just be something like this instead?

```cpp
UIRangeAnnulusChangedEvent( const RangeAnnulus Annulus )
	: Annulus( Annulus )
{
}
```

My initial thinking was that it'd be nice to just pass in the whole `RangeAnnulusChangedEvent` and keep the behavior that translates between the two in UIRangeAnnulusChangedEvent.h (as currently seen in the commit), but I guess I can see the case for maybe wanting to keep them uncoupled (following the proposed snippet above)?

Let me know what you think!

Reply:
I think it's fine as is - it's a direct event translation so I think either works but this is more nicely coupled to imply that they should probably match.